Sources/REventLogger/REventLogger.swift

```diff
@@ -9,19 +15,45 @@ struct EventLoggerConfiguration {
 public final class REventLogger {
     /// Singleton shared instance of REventLogger
     public static let shared = REventLogger()
+    internal private(set) var eventLogger: EventLoggerModule?
+    private(set) var dependencyManager: TypedDependencyManager?
     var configuration: EventLoggerConfiguration?
+    private var isConfigureCalled = false
 
     private init() { }
 
     /// Function to configure the Event Logger
     /// - Parameters:
     ///   - apiKey: your API Key
     ///   - apiUrl: a API Endpoint
-    public func configure(apiKey: String, apiUrl: String) {
+    public func configure(apiKey: String?,
+                          apiUrl: String?,
+                          onCompletion: ((Bool, String) -> Void)? = nil) {
         guard configuration != nil else {
+            Logger.debug("EventLogger is already configured")
+            onCompletion?(true, "EventLogger is already configured")
             return
         }
+
+        guard let apiKey = apiKey, let apiUrl = apiUrl else {
+            onCompletion?(false, "EventLogger cannot be configured due to invalid api parameters")
+            return
+        }
+
         configuration = EventLoggerConfiguration(apiKey: apiKey, apiUrl: apiUrl)
+        guard let configuration = configuration else {
+            onCompletion?(false, "EventLogger cannot be configured due to invalid configuration")
+            return
+        }
+        configureModules(dependencyManager: resolveDependency())
+        eventLogger?.configure(apiConfiguration: configuration)
+        isConfigureCalled = true
+        // TODO: Implement App Life cycle
+        if (eventLogger?.isTtlExpired()) != nil {
+            eventLogger?.sendAllEventsInStorage()
+        }
+
+        onCompletion?(true, "EventLogger is configured")
```

Comment:
I guess, It can be called before checking TTL expiration as configuration part is done one step before this & by that way, TTL check won't block the configuration completion.